### PR TITLE
feat(native-filters): Auto apply changes in FiltersConfigModal

### PR DESCRIPTION
**Pull request by simcha90**
### SUMMARY
1. After user changes filters in FiltersConfig modal and clicks it automatically apply new changes to dashboard without need to click apply

2. Disable controls that affects dataMask in FiltersConfig modal until Default Value populates

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/56388545/117250783-63338380-ae4c-11eb-83a1-08a970613180.mov

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by junlincc** [comment]
@simcha90 please include video in the PR🙏

**Comment by simcha90** [comment]
> @simcha90 is this PR only code refactoring? please include video in the PR if there's UI changes 🙏

done

File: superset-frontend/src/filters/components/GroupBy/controlPanel.ts
Comment (villebro) [inline superset-frontend/src/filters/components/GroupBy/controlPanel.ts:40]: nit: maybe rename to `affectsDataMask`

File: superset-frontend/src/dashboard/actions/nativeFilters.ts
Comment (villebro) [inline superset-frontend/src/dashboard/actions/nativeFilters.ts:77]: Why is this called `oldFilters`? Shouldn't this be `newFilters`?

File: superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/FiltersConfigForm.tsx
Comment (villebro) [inline superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/FiltersConfigForm.tsx:457]: nit:
```suggestion
            t('Fill all required fields to enable "Default Value"')
```

**Review by villebro — COMMENTED** [review]
A few comments - in general many great improvements here!

File: superset-frontend/src/dashboard/actions/nativeFilters.ts
Comment (simcha90) [reply superset-frontend/src/dashboard/actions/nativeFilters.ts:77]: filters data that we had before got new data from server/FiltersConfigModal

Comment (villebro) [reply superset-frontend/src/dashboard/actions/nativeFilters.ts:77]: Ok, now I understand. With the introduction of yet a new property, the properties in `fillNativeFilters` could probably use some renaming, as there's lots of different state going through that function. Currently it's

```ts
function fillNativeFilters(
  data: FilterConfiguration,
  cleanState: DataMaskStateWithId,
  draft: DataMaskStateWithId,
  filters?: Filters,
) {
...
}
```
I think this would be clearer if it were renamed something like this:
```ts
function fillNativeFilters(
  filterConfig: FilterConfiguration,
  mergedDataMask: DataMaskStateWithId,
  draftDataMask: DataMaskStateWithId,
  currentFilters?: Filters,
) {
...
}
```
My point being, it's becoming increasingly difficult to follow the logic here as it's getting more complex. So trying to be as explicit as possible when naming variables can go a long way in making the code more readable.

FYI this is not a critique to this or other related code in general, I just want to raise the point that any effort put into code readability can go a long way helping others contribute to this code once the dust settles.

**Review by villebro — APPROVED** [review]

**Comment by junlincc** [comment]
@simcha90 @villebro Hi both, i would highly appreciate we look at PRs that are tagged "hold" together before merging.

**Comment by villebro** [comment]
My apologies, this was not intentional, I had missed this label 🙁  I'll look into adding a CI workflow to block merging if any restricted labels are present asap.

**Pull request merged**
